src/solvers/refinement/string_refinement.cpp

```
@@ -2534,16 +2534,16 @@ exprt string_refinementt::get(const exprt &expr) const
       auto set = generator.get_created_strings();
       if(set.find(arr) != set.end())
       {
-        exprt length = super_get(arr.length());
-        if(const auto n = numeric_cast<std::size_t>(length))
+        const exprt length = super_get(arr.length());
+        const auto n = numeric_cast<std::size_t>(length);
+        if(n.has_value() && *n <= max_string_length)
```

Is it correct to make this fail silently?